### ISSUE TITLE
Where to find evaluation results

### ISSUE THREAD

**Issue by AntoineCribel** [DeepLabCut/DeepLabCut#707]
**Is your feature request related to a problem? Please describe.**
With the new update (2.2), I don't now anymore where to find the mean pixel error when evaluating a network. It doesn't appears in the terminal anymore and there is no .csv file in the evaluation-results folder. I also tried running dlc.return_evaluate_network_data() but it only return the path of a hdf5 file (which doesn't seem to contain any pixel errors).

**Describe the solution you'd like**
Would be nice to get similar pixel error than in previous deeplabcut version to be able to compare the new version (using the skeleton) to previously trained networks. Currently the evaluation procedure seems quite confusing to me.

**Comment by HenryDarch** [comment]
I also have the same issue with the new beta release.

**Comment by jeylau** [comment]
@AntoineCribel, @HenryDarch, the evaluation now requires an additional, critical stage of cross-validation ([https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/UseOverviewGuide.md#evaluate-the-trained-network](https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/UseOverviewGuide.md#evaluate-the-trained-network). Something like
`deeplabcut.evaluate_multianimal_crossvalidate(config_path, Shuffles=[1], edgewisecondition=True, leastbpts=1, init_points=20, n_iter=50)` is a great start.
It is only after the best inference parameters are found that you will get the results in the terminal + the .csv file (which will also contains extra metrics such as PCK or the number of misses) 😃

**Comment by HenryDarch** [comment]
Thanks for the clarification @jeylau. That brings me onto my error (sorry Antoine for hijacking this thread a bit, perhaps you will have the same issue?):- 
When running the X-validation code (after the initial evaluation) an error is thrown that seems to traceback to a missing 'inferencebounds.yaml' file, screenshot below.

code run: 
`deeplabcut.evaluate_network(config_file, plotting=True, show_errors=True)
deeplabcut.evaluate_multianimal_crossvalidate(config_file, Shuffles=[1], edgewisecondition=True, leastbpts=1, init_points=20, n_iter=50)


![image](https://user-images.githubusercontent.com/54323442/83013513-0f51d500-a015-11ea-83df-a3ab4b6e9a3a.png)

**Comment by MMathisLab** [comment]
@AntoineCribel - the stats are printed in the terminal after x-eval; Please see the docs here:  https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/functionDetails.md#cross-validation-of-inference-parameters-a-madeeplabcut-critical-point

```
Saving optimal inference parameters...
   train_iter  train_frac  shuffle  rmse_train  hits_train  misses_train  falsepos_train  ndetects_train  pck_train  rpck_train  rmse_test  hits_test  misses_test  falsepos_test  ndetects_test  pck_test  rpck_test
0     50000.0        95.0      1.0   36.681365     9.89759     11.645783         0.63494        1.761446   0.306809    0.288352   33.81332      8.675       13.025           0.45          1.625  0.286614   0.264459
```

Also, @AntoineCribel - please definitely check out the video tutorials, they are only ~10 min but take you through the whole process. Multi-animal is more complicated, but I tried hard to make it as straight forward as possible with the GUI, docs, and videos :) 

https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/UseOverviewGuide.md#important-information-on-using-deeplabcut

**Comment by MMathisLab** [comment]
@HenryDarch can you provide the full traceback, please.

**Comment by HenryDarch** [comment]
Hi @MMathisLab, of course.
System specs for clarity
OS: Ubuntu 18.04, CUDA 10.0, Cudnn 7.4.2, tf 1.13.1
GPU: RTX 2070 8GB
code run: `deeplabcut.evaluate_multianimal_crossvalidate(config_file, Shuffles=[1], edgewisecondition=True, leastbpts=1, init_points=20, n_iter=50)`

Traceback:
```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
~/anaconda3/envs/DLC/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/lib/crossvalutils.py in bayesian_search(config_path, inferencecfg, pbounds, edgewisecondition, shuffle, trainingsetindex, modelprefix, snapshotindex, target, maximize, init_points, n_iter, acq, log_file, dcorr, leastbpts, printingintermediatevalues)
    370             inferenceboundscfg = auxiliaryfunctions.read_plainconfig(
--> 371                 path_inferencebounds_config
    372             )

~/anaconda3/envs/DLC/lib/python3.7/site-packages/deeplabcut/utils/auxiliaryfunctions.py in read_plainconfig(configname)
    262         raise FileNotFoundError(
--> 263             f"Config {configname} is not found. Please make sure that the file exists."
    264         )

FileNotFoundError: Config /home/mccaffertylab/Documents/DLC-projects/4_Mouse_OFT-Katie-2020-05-24/dlc-models/iteration-0/4_Mouse_OFTMay24-trainset95shuffle1/test/inferencebounds.yaml is not found. Please make sure that the file exists.

During handling of the above exception, another exception occurred:

KeyError                                  Traceback (most recent call last)
<ipython-input-6-d51fd8d422f9> in <module>
----> 1 deeplabcut.evaluate_multianimal_crossvalidate(config_file, Shuffles=[1], edgewisecondition=True, leastbpts=1, init_points=20, n_iter=50)

~/anaconda3/envs/DLC/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/evaluate_multianimal.py in evaluate_multianimal_crossvalidate(config, Shuffles, trainingsetindex, pbounds, edgewisecondition, target, inferencecfg, init_points, n_iter, dcorr, leastbpts, printingintermediatevalues, modelprefix, plotting)
    565             dcorr=dcorr,
    566             leastbpts=leastbpts,
--> 567             modelprefix=modelprefix,
    568         )
    569 

~/anaconda3/envs/DLC/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/lib/crossvalutils.py in bayesian_search(config_path, inferencecfg, pbounds, edgewisecondition, shuffle, trainingsetindex, modelprefix, snapshotindex, target, maximize, init_points, n_iter, acq, log_file, dcorr, leastbpts, printingintermediatevalues)
    376 
    377             inferenceboundscfg = calculatepafdistancebounds(
--> 378                 config_path, shuffle, trainingsetindex
    379             )
    380             auxiliaryfunctions.write_plainconfig(

~/anaconda3/envs/DLC/lib/python3.7/site-packages/deeplabcut/pose_estimation_tensorflow/evaluate.py in calculatepafdistancebounds(config, shuffle, trainingsetindex, modelprefix, numdigits, onlytrain)
    138                     if ind != "single" and ind2 != "single":
    139                         distances = np.sqrt(
--> 140                             (Data[ind, j1, "x"] - Data[ind2, j1, "x"]) ** 2
    141                             + (Data[ind, j1, "y"] - Data[ind2, j2, "y"]) ** 2
    142                         )

~/anaconda3/envs/DLC/lib/python3.7/site-packages/pandas/core/frame.py in __getitem__(self, key)
   2797         if is_single_key:
   2798             if self.columns.nlevels > 1:
-> 2799                 return self._getitem_multilevel(key)
   2800             indexer = self.columns.get_loc(key)
   2801             if is_integer(indexer):

~/anaconda3/envs/DLC/lib/python3.7/site-packages/pandas/core/frame.py in _getitem_multilevel(self, key)
   2847     def _getitem_multilevel(self, key):
   2848         # self.columns is a MultiIndex
-> 2849         loc = self.columns.get_loc(key)
   2850         if isinstance(loc, (slice, Series, np.ndarray, Index)):
   2851             new_columns = self.columns[loc]

~/anaconda3/envs/DLC/lib/python3.7/site-packages/pandas/core/indexes/multi.py in get_loc(self, key, method)
   2670 
   2671         if keylen == self.nlevels and self.is_unique:
-> 2672             return self._engine.get_loc(key)
   2673 
   2674         # -- partial selection or non-unique index

pandas/_libs/index.pyx in pandas._libs.index.BaseMultiIndexCodesEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/index.pyx in pandas._libs.index.IndexEngine.get_loc()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.UInt64HashTable.get_item()

pandas/_libs/hashtable_class_helper.pxi in pandas._libs.hashtable.UInt64HashTable.get_item()

KeyError: 209
```

**Comment by MMathisLab** [comment]
thanks! can you confirm this file is indeed there? 
`/home/mccaffertylab/Documents/DLC-projects/4_Mouse_OFT-Katie-2020-05-24/dlc-models/iteration-0/4_Mouse_OFTMay24-trainset95shuffle1/test/inferencebounds.yaml`

and that in dlc-models, .. you have snapshots?

**Comment by HenryDarch** [comment]
Thanks for the quick reply.
There is no 'inferencebounds.yaml' file in any subdirectory (there is 'inference_cfg.yaml' under the "models/iteration/name/test" folder). Is this something that the user needs to create? If so I apologise and must have missed that in the docs.

dlc-models sub-dir has all the expected snapshots, as well as 'pose_cfg.yaml', 'log.txt','learning_stats.csv', 'checkpoint', and a log folder.

**Comment by AlexEMG** [comment]
For multi animal projects, when one creates a training set `inference_cfg.yaml `will be created automatically along with `pose_cfg.yaml` based on default variables in https://github.com/DeepLabCut/DeepLabCut/blob/master/deeplabcut/inference_cfg.yaml

The user can of course edit those values, and cross-validation during evaluation will also update them.

**Comment by MMathisLab** [comment]
Thanks @HenryDarch - okay, odd this is happening. Typically it is created at the evaluation step, and found here:
![Screen Shot 2020-05-27 at 4 59 42 PM](https://user-images.githubusercontent.com/28102185/83071758-a03ba700-a03b-11ea-8245-cadf2cf7734a.png)

Could you re-run the "evaluate" step, and then x-val again? (and if you are on Windows, be sure you open the terminal "as an admin" just in case some funny permission issue with writing the file)

**Comment by HenryDarch** [comment]
No change when pre-running evaluation. I'll try re-creating the project, perhaps in a fresh environment. 
Feel free to close this as the original question has been answered. I'll create a dedicated issue if my bug persists. :)

**Comment by AntoineCribel** [comment]
Thanks @jeylau and @MMathisLab!! 
I still think it would be nice to have a .csv file with those metrics though.

**Comment by MMathisLab** [comment]
@AntoineCribel - you simply need to pass save as csv to true; check out the docstring. There is an h5 file by default, which is just a more compact format ;) and readable in python, R, matlab, Colab, etc.
